src/coreComponents/finiteElement/unitTests/testFiniteElementBase.cpp

```diff
@@ -117,23 +115,596 @@ TEST( FiniteElementBase, testJacobianInverseHost )
+
+  for( int q=0; q<20; ++q )
+  {
+    randomShape( N );
+    randomVar( f );
+
+    plus_NaFi( N, f, baselineResult );
+    FiniteElementBase::plus_NaFi( N, f, feResult );
+  }
+
+  for( int a=0; a<NUM_SUPPORT_POINTS; ++a )
+  {
+    for( int i=0; i<3; ++i )
+    {
+      EXPECT_FLOAT_EQ( feResult[a][i], baselineResult[a][i] );
+    }
+  }
+}
+
+
+
+//***** TEST plus_gradNajAij_plus_NaFi() **********************************************************
+
+
+template< int NUM_SUPPORT_POINTS >
+static void plus_gradNajAij_plus_NaFi( real64 const (&gradN)[NUM_SUPPORT_POINTS][3],
+                                       real64 const (&var_detJxW)[6],
+                                       real64 const (&N)[NUM_SUPPORT_POINTS],
+                                       real64 const (&forcingTerm_detJxW)[3],
+                                       real64 (& R)[NUM_SUPPORT_POINTS][3] )
+{
+  for( int a=0; a<NUM_SUPPORT_POINTS; ++a )
+  {
+    R[a][0] += var_detJxW[0] * gradN[a][0] + var_detJxW[5] * gradN[a][1] + var_detJxW[4] * gradN[a][2] + forcingTerm_detJxW[0] * N[a];
+    R[a][1] += var_detJxW[5] * gradN[a][0] + var_detJxW[1] * gradN[a][1] + var_detJxW[3] * gradN[a][2] + forcingTerm_detJxW[1] * N[a];
+    R[a][2] += var_detJxW[4] * gradN[a][0] + var_detJxW[3] * gradN[a][1] + var_detJxW[2] * gradN[a][2] + forcingTerm_detJxW[2] * N[a];
+  }
+}
+
+template< int NUM_SUPPORT_POINTS >
+static void plus_gradNajAij_plus_NaFi( real64 const (&gradN)[NUM_SUPPORT_POINTS][3],
+                                       real64 const (&var_detJxW)[3][3],
+                                       real64 const (&N)[NUM_SUPPORT_POINTS],
+                                       real64 const (&forcingTerm_detJxW)[3],
+                                       real64 (& R)[NUM_SUPPORT_POINTS][3] )
+{
+  for( int a=0; a<NUM_SUPPORT_POINTS; ++a )
+  {
+    R[a][0] += var_detJxW[0][0] * gradN[a][0] + var_detJxW[0][1] * gradN[a][1] + var_detJxW[0][2] * gradN[a][2] + forcingTerm_detJxW[0] * N[a];
+    R[a][1] += var_detJxW[1][0] * gradN[a][0] + var_detJxW[1][1] * gradN[a][1] + var_detJxW[1][2] * gradN[a][2] + forcingTerm_detJxW[1] * N[a];
+    R[a][2] += var_detJxW[2][0] * gradN[a][0] + var_detJxW[2][1] * gradN[a][1] + var_detJxW[2][2] * gradN[a][2] + forcingTerm_detJxW[2] * N[a];
+  }
+}
+
+TEST( FiniteElementBase, test_plus_gradNajAij_plus_NaFi )
+{
+  srand( 1234 );
+  constexpr int NUM_SUPPORT_POINTS = 9;
+  real64 N[NUM_SUPPORT_POINTS];
+  real64 gradN[NUM_SUPPORT_POINTS][3];
+
+  real64 r2Tensor[3][3];
+  real64 r2SymmTensor[6];
+  real64 f[3];
+
+  real64 baselineResult[NUM_SUPPORT_POINTS][3] = {{0}};
+  real64 feResult[NUM_SUPPORT_POINTS][3] = {{0}};
+  real64 baselineResultSym[NUM_SUPPORT_POINTS][3] = {{0}};
+  real64 feResultSym[NUM_SUPPORT_POINTS][3] = {{0}};
+
+  for( int q=0; q<20; ++q )
+  {
+    randomShape( N );
+    randomShapeGradient( gradN );
+    randomVar( r2Tensor );
+    randomVar( r2SymmTensor );
+    randomVar( f );
```

Comment:
We really should be using `std::uniform_real_distribution` instead of `rand` and `srand`.

Reply:
yeah. I really doesn't matter how the distribution is. I just trying to fill the values with non-nice numbers.

Reply:
Sure I just meant we should move away from `rand` to the std random library.

Reply:
Oh. I totally agree. And you are right in that I should just use the newer approach in case someone copies this somewhere it matters.